### PR TITLE
refactor(iceberg): extract IcebergCommon config

### PR DESCRIPTION
**Pull request by xxchan**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

Previously, we use `sink/IcebergConfig` for both source and sink, which is confusing. This PR 
- separates out common config and methods.
- Don't convert `source::IcerbergProperties` to `sink::IcebergConfig`

Note: currently source and sink has some differences:
- source doesn't support path_style_access
- source supports less options in java_catalog_props

This PR tries to be an equivalent refactor and so doesn't change any of these.

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

**Comment by xxchan** [comment]
* **#18600** <a href="https://app.graphite.dev/github/pr/risingwavelabs/risingwave/18600?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @xxchan and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: src/connector/src/connector_common/iceberg/mod.rs [hunk 1]
```diff
@@ -0,0 +1,592 @@
...
+    }
+
+    pub fn catalog_name(&self) -> String {
+        self.catalog_name
+            .as_ref()
+            .map(|s| s.to_string())
+            .unwrap_or_else(|| "risingwave".to_string())
+    }
+
+    pub fn full_table_name(&self) -> ConnectorResult<TableIdentifier> {
```
Comment (Li0k) [inline src/connector/src/connector_common/iceberg/mod.rs:82]: how about move `full_table_name` to v1 ? It use icelake::TableIdentifier

**Review by Li0k — COMMENTED** [review]
Rest LGTM, thanks for the efforts

File: src/connector/src/connector_common/iceberg/mod.rs [hunk 2]
```diff
@@ -0,0 +1,595 @@
...
+    #[serde(rename = "catalog.type")]
+    pub catalog_type: Option<String>,
+    #[serde(rename = "s3.region")]
+    pub region: Option<String>,
+    #[serde(rename = "s3.endpoint")]
+    pub endpoint: Option<String>,
+    #[serde(rename = "s3.access.key")]
+    pub access_key: String,
+    #[serde(rename = "s3.secret.key")]
+    pub secret_key: String,
```
Comment (tabVersion) [inline src/connector/src/connector_common/iceberg/mod.rs:49]: referencing this part to `AwsAuthProps` instead?

**Review by tabVersion — APPROVED** [review]
Are both source and sink using `IcebergProperties`?

overall LGTM

File: src/connector/src/connector_common/iceberg/mod.rs (hunk 2 above)
Comment (xxchan) [reply src/connector/src/connector_common/iceberg/mod.rs:49]: They are incompatible and this PR doesn't want to change behavior

**Review by chenzl25 — APPROVED** [review]
LGTM, thanks!

**Pull request merged**